### PR TITLE
Fixed testparamter in ZipArchiverTest

### PR DESCRIPTION
**Pull request by StefanSpieker**
I stumbled upon a minor issue I thought, but this might be a real bug in the implementation. The existing txt file was just 2 bytes big, instead of 4gb +2. As soon as I corrected this, the test fails.

The PR introducing this functionality was: https://github.com/jenkinsci/jenkins/pull/3536

### Proposed changelog entries

* TBD

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@jsoref 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by StefanSpieker** [comment]
I tried setting the to `Zip64Mode.Always` which fixed the writing part but broke the reading ...

**Comment by jsoref** [comment]
Off-topic, you're missing a letter near: `paramter`.

But, is this just converting a broken test into a failing test as opposed to actually getting the system to pass?

(Thanks for spotting it either way... "oops" on my part.)

**Comment by jsoref** [comment]
This:
https://github.com/jenkinsci/jenkins/blob/360029e9d52152425dbabca9f1072fcd919772b6/core/src/test/java/hudson/util/io/ZipArchiverTest.java#L66
... probably should be a `ZipOutputStream`, as in:
https://github.com/jenkinsci/jenkins/blob/dcc9a924d84d2d35166166dde0ebfe2e9ef52ab0/core/src/test/java/hudson/PluginManagerTest.java#L158

**Comment by StefanSpieker** [comment]
I'm not really sure if the test is broken or the code is not working. Are you sure the code works?

File: core/src/test/java/hudson/util/io/ZipArchiverTest.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ public void huge64bitFile()  {
         File hugeFile = new File(tmpDir, "huge64bitFileTest.txt");
         try {
             RandomAccessFile largeFile = new RandomAccessFile(hugeFile, "rw");
-            largeFile.setLength(4 * 1024 * 1024 * 1024 + 2);
+            largeFile.setLength(4L * 1024 * 1024 * 1024 + 2);
```
Comment (MRamonLeon) [inline core/src/test/java/hudson/util/io/ZipArchiverTest.java:102]: That's right, `System.out.println(4 * 1024 * 1024 * 1024 + 2);` prints 2
The file used is indeed 2 bytes lenght.

@oleg-nenashev maybe this is affecting here:
https://commons.apache.org/proper/commons-compress/javadocs/api-1.18/org/apache/commons/compress/archivers/zip/ZipArchiveOutputStream.html#setUseZip64-org.apache.commons.compress.archivers.zip.Zip64Mode-

> When setting the mode to AsNeeded, Zip64 extensions will transparently be used for those entries that require them. This mode can only be used if the uncompressed size of the ZipArchiveEntry is known when calling putArchiveEntry(org.apache.commons.compress.archivers.ArchiveEntry) or the archive is written to a seekable output (i.e. you have used the File-arg constructor) - this mode is not valid when the output stream is not seekable and the uncompressed size is unknown when putArchiveEntry(org.apache.commons.compress.archivers.ArchiveEntry) is called.

File: core/src/test/java/hudson/util/io/ZipArchiverTest.java [hunk 2]
```diff
@@ -135,7 +135,6 @@ public void huge64bitFile()  {
         String zipEntryName = null;
 
         try (ZipFile zipFileVerify = new ZipFile(zipFile)) {
-
```
Comment (MRamonLeon) [inline core/src/test/java/hudson/util/io/ZipArchiverTest.java:138]: avoid adding unrelated changes to your PR

**Review by MRamonLeon — COMMENTED** [review]
Indeed, maybe we need to fix the feature

**Comment by jsoref** [comment]
Fwiw, here's what I have:
https://github.com/jsoref/jenkins/commit/509a2a15b9911ed032814cdd753505a6ed8e3fc5

I looked into the other side and stopped.
